src/ArrayList.java

```diff
@@ -0,0 +1,73 @@
+package ArrListLinkedList;
+
+public class ArrayList<T> implements List<T> {
+    private static final int LIST_SIZE_BY_DEFAULT = 10;
+    private Object[] val;
+    private int size = 0;
+    private int index;
+
+    public ArrayList() {
+        this.val = new Object[LIST_SIZE_BY_DEFAULT];
+        this.size = LIST_SIZE_BY_DEFAULT;
+    }
+
+    @Override
+    public T get(int i) throws Exception {
+        if (i > this.index - 1) {
```

Comment:
you should create helper method which check index and throw indexOfBoundException with current size and index